book/types_of_data.md

```diff
@@ -395,11 +396,30 @@ To remove rows from a table, you'll commonly use the `select` command with row n
 
 There are numerous other commands for selecting and reducing the data in tables, records and lists.
 
+## Closures
+
+Closures are anonymous functions that can close over variables outside their scope and can be passed as a value.
+For example, in the command `each { |it| print $it }` the closure is the portion contained in curly braces, `{ |it| print $it }`.
+Closure parameters are specified between a pair of pipe symbols (for example, `|it|`) if necessary.
+You can also use `$in` in most closures instead of providing a parameter: `each { print $in }`
```

Comment:
Mhh `$in` semantics are always a bit confusing. Would be great if we could specify the rules when it is available in the future. 

I think in this case we should make clear that this is not traditional closing over the variable.

```suggestion
You can also use a pipeline input as `$in` in most closures instead of providing an explicit parameter: `each { print $in }`
```

Reply:
Agreed.